src/native/src/components/Badge/components/BadgeIcon/index.tsx

```diff
@@ -0,0 +1,19 @@
+import { useTheme } from "styled-components/native";
+import { BadgeChildrenProps } from "../../types";
+import Icon, { IconKeys } from "../../../Icon";
+import { getBadgeColors } from "../../Colors";
+import { useComponentProps } from "../../../../providers/ComponentPropsProvider";
+
+export interface BadgeIconProps {
+  name: IconKeys;
+}
+const BadgeIcon = ({ name }: BadgeIconProps) => {
+  const theme = useTheme();
+  const { color, isOutline } = useComponentProps<BadgeChildrenProps>();
+
+  const iconColor = getBadgeColors(color, theme, isOutline).icon;
+
+  return <Icon color={iconColor} size={1} name={name} />;
+};
```

Comment:
Tava pensando em a gente tenta uma solução pra remover esse tanto de prefixo/sufixo `Badge` pq td q ta aqui dentro já faz parte do componente `Badge` meio q ta redundante, sei la, me agonia um pouco kkkk, que tal? Só n pensei numa ideia boa pro componente `Icon`
 
```suggestion
import DefaultIcon, { IconKeys } from "../../../Icon";
import { getColors } from "../../Colors";
import { useComponentProps } from "../../../../providers/ComponentPropsProvider";

export interface IconProps {
  name: IconKeys;
}
const Icon = ({ name }: IconProps) => {
  const theme = useTheme();
  const { color, isOutline } = useComponentProps<BadgeProps>();

  const colors = getColors(color, theme, isOutline);

  return <DefaultIcon color={colors.icon} size={1} name={name} />;
};
```